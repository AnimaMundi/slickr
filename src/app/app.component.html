<header><app-header></app-header></header>

<main>
  <app-search-form-container></app-search-form-container>

  <app-result-list-container></app-result-list-container>

  <router-outlet></router-outlet>
</main>

<footer><app-footer></app-footer></footer>
