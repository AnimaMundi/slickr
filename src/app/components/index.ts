export * from './footer';
export * from './header';
export * from './result-list';
export * from './result-list-container';
export * from './search-form';
export * from './search-form-container';
