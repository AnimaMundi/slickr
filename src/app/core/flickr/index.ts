export * from './models';
export * from './flickr.service';
