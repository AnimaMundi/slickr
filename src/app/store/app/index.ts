export * from './app.reducer';
export * from './app.state';
export * from './router-state-serializer.class';
