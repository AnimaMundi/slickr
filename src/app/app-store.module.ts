import { NgModule } from '@angular/core';

import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';
import { StoreRouterConnectingModule } from '@ngrx/router-store';

import { appReducer, metaReducers, CustomSerializer } from '@store/app';
import { SearchEffects } from '@store/search';
import { environment } from '@env';

@NgModule({
  imports: [
    StoreModule.forRoot(appReducer, { metaReducers }),
    !environment.production ? StoreDevtoolsModule.instrument() : [],
    EffectsModule.forRoot([SearchEffects]),
    StoreRouterConnectingModule.forRoot({
      serializer: CustomSerializer,
    }),
  ],
})
export class AppStoreModule {}
