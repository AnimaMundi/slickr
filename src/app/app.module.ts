import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule, FormsModule } from '@angular/forms';
import { HttpClientModule } from '@angular/common/http';

import {
  MatToolbarModule,
  MatFormFieldModule,
  MatInputModule,
  MatIconModule,
  MatCardModule,
  MatButtonModule,
} from '@angular/material';

import { AppRoutingModule } from './app-routing.module';
import { AppStoreModule } from './app-store.module';

import { AppComponent } from './app.component';
import {
  HeaderComponent,
  FooterComponent,
  ResultListComponent,
  ResultListContainerComponent,
  SearchFormComponent,
  SearchFormContainerComponent,
} from './components';

@NgModule({
  declarations: [
    AppComponent,
    HeaderComponent,
    FooterComponent,
    ResultListComponent,
    ResultListContainerComponent,
    SearchFormComponent,
    SearchFormContainerComponent,
  ],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    ReactiveFormsModule,
    FormsModule,
    HttpClientModule,

    MatToolbarModule,
    MatFormFieldModule,
    MatInputModule,
    MatIconModule,
    MatCardModule,
    MatButtonModule,

    AppRoutingModule,
    AppStoreModule,
  ],
  bootstrap: [AppComponent],
})
export class AppModule {}
