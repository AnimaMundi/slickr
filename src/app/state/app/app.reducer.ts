import { ActionReducerMap, MetaReducer } from '@ngrx/store';
import { routerReducer } from '@ngrx/router-store';

import { environment } from '../../../environments/environment';
import { AppState } from './app.state';

export const appReducer: ActionReducerMap<AppState> = {
  router: routerReducer,
};

export const metaReducers: MetaReducer<AppState>[] = !environment.production
  ? []
  : [];
