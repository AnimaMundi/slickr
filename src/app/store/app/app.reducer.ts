import { ActionReducerMap, MetaReducer } from '@ngrx/store';
import { routerReducer } from '@ngrx/router-store';

import { environment } from '../../../environments/environment';
import { AppState } from './app.state';
import { searchReducer, searchStateKey } from '../search';

export const appReducer: ActionReducerMap<AppState> = {
  router: routerReducer,
  [searchStateKey]: searchReducer,
};

export const metaReducers: MetaReducer<AppState>[] = !environment.production
  ? []
  : [];
