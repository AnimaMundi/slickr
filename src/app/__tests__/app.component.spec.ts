import { NO_ERRORS_SCHEMA } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';

import { AppComponent } from '../app.component';

describe('AppComponent', () => {
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [AppComponent],
      schemas: [NO_ERRORS_SCHEMA],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should compile the template', () => {
    expect(fixture).toMatchSnapshot();
  });
});
