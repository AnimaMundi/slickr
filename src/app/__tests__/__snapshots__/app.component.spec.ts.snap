// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should compile the template 1`] = `
<app-root>
  <header>
    <app-header />
  </header><main>
    <app-search-form-container />
    <app-result-list-container />
    <router-outlet />
  </main><footer>
    <app-footer />
  </footer>
</app-root>
`;
